agent/lib_aws_sdk_php.c

```diff
@@ -519,6 +518,171 @@ char* nr_lib_aws_sdk_php_get_command_arg_value(char* command_arg_name,
   return command_arg_value;
 }
 
+void nr_lib_aws_sdk_php_dynamodb_set_params(
+    nr_segment_datastore_params_t* datastore_params,
+    nr_segment_cloud_attrs_t* cloud_attrs,
+    NR_EXECUTE_PROTO) {
+  zval* endpoint_zval = NULL;
+  zval* region_zval = NULL;
+  zval* host_zval = NULL;
+  zval* port_zval = NULL;
+  zval* this_obj = NULL;
+  zend_function* func = NULL;
+  zend_class_entry* base_class = NULL;
+  char* table_name = NULL;
+  char* account_id = NULL;
+
+  if (NULL == datastore_params || NULL == cloud_attrs) {
+    return;
+  }
+
+  this_obj = NR_PHP_USER_FN_THIS();
+  func = nr_php_execute_function(NR_EXECUTE_ORIG_ARGS);
+
+  if (NULL == this_obj || NULL == func) {
+    return;
+  }
+
+  if (NULL != func->common.scope) {
+    base_class = func->common.scope;
+
+    region_zval = nr_php_get_zval_object_property_with_class(
+        this_obj, base_class, "region");
+    if (nr_php_is_zval_non_empty_string(region_zval)) {
+      cloud_attrs->cloud_region = Z_STRVAL_P(region_zval);
+    }
+
+    endpoint_zval = nr_php_get_zval_object_property_with_class(
+        this_obj, base_class, "endpoint");
+    if (nr_php_is_zval_valid_object(endpoint_zval)) {
+      host_zval = nr_php_get_zval_object_property(endpoint_zval, "host");
+      if (nr_php_is_zval_non_empty_string(host_zval)) {
+        datastore_params->instance->host = Z_STRVAL_P(host_zval);
+
+        /* Only try to get a port if we have a valid host. */
+        port_zval = nr_php_get_zval_object_property(endpoint_zval, "port");
+        if (nr_php_is_zval_valid_integer(port_zval)) {
+          /* Must be freed by caller */
+          datastore_params->instance->port_path_or_id
+              = nr_formatf(NR_INT64_FMT, Z_LVAL_P(port_zval));
+        } else {
+          /* In case where host was found but port was not, spec says return
+           * unknown for port. */
+          datastore_params->instance->port_path_or_id = nr_strdup("unknown");
```

Comment:
Would this just be the default port if unset?

Reply:
No because we were able to extract a host just not a port.  Once we've extracted a host, we cannot assume a port to go with it.  Default port only ever pairs with default host.